browser/android/brave_sync_worker.cc

```diff
@@ -239,122 +239,58 @@ void BraveSyncWorker::OnResetDone() {
   }
 }
 
-namespace {
-
-// A structure which contains all the configuration information for sync.
-struct SyncConfigInfo {
-  SyncConfigInfo();
-  ~SyncConfigInfo();
-
-  bool encrypt_all;
-  std::string passphrase;
-  bool set_new_passphrase;
-};
-
-SyncConfigInfo::SyncConfigInfo()
-    : encrypt_all(false), set_new_passphrase(false) {}
-
-SyncConfigInfo::~SyncConfigInfo() {}
-
-// Return false if we are not interested configure encryption
-bool FillSyncConfigInfo(syncer::SyncService* service,
-                        SyncConfigInfo* configuration,
-                        const std::string& passphrase) {
-  bool first_setup_in_progress =
-      service && !service->GetUserSettings()->IsFirstSetupComplete();
-
-  configuration->encrypt_all =
-      service->GetUserSettings()->IsEncryptEverythingEnabled();
-
-  bool sync_prefs_passphrase_required =
-      service->GetUserSettings()->IsPassphraseRequired();
-
-  if (!first_setup_in_progress) {
-    if (!configuration->encrypt_all) {
-      configuration->encrypt_all = true;
-      configuration->set_new_passphrase = true;
-      DCHECK_NE(passphrase.size(), 0u);
-      configuration->passphrase = passphrase;
-    } else if (sync_prefs_passphrase_required) {
-      configuration->set_new_passphrase = false;
-      DCHECK_NE(passphrase.size(), 0u);
-      configuration->passphrase = passphrase;
-    } else {
-      return false;
-    }
+void BraveSyncWorker::SetEncryptionPassphrase() {
+  syncer::SyncService* service = GetSyncService();
+  DCHECK(service);
+  DCHECK(service->IsEngineInitialized());
+  DCHECK(!this->passphrase_.empty());
+
+  syncer::SyncUserSettings* sync_user_settings =
+      GetSyncService()->GetUserSettings();
+  DCHECK(!sync_user_settings->IsPassphraseRequired());
+
+  if (sync_user_settings->IsEncryptEverythingAllowed() &&
+      !sync_user_settings->IsUsingSecondaryPassphrase() &&
+      !sync_user_settings->IsTrustedVaultKeyRequired()) {
+    sync_user_settings->SetEncryptionPassphrase(this->passphrase_);
+    ProfileMetrics::LogProfileSyncInfo(
+        ProfileMetrics::SYNC_CREATED_NEW_PASSPHRASE);
   }
-  return true;
 }
 
-}  // namespace
+void BraveSyncWorker::SetDecryptionPassphrase() {
+  syncer::SyncService* service = GetSyncService();
+  DCHECK(service);
+  DCHECK(service->IsEngineInitialized());
+  DCHECK(!this->passphrase_.empty());
+  syncer::SyncUserSettings* sync_user_settings =
+      GetSyncService()->GetUserSettings();
```

Comment:
Should we not use `service` instead of `GetSyncService()`

Reply:
thanks, fixed at 2bdf046a084cb5445c71d45ef28cfd7cbcab6221